frontend/components/shelter/home/PetsTable/components/Pill.tsx

```diff
@@ -0,0 +1,59 @@
+import { Tag } from "antd";
+import {
+	PillColor,
+	Species,
+	SpeciesTags,
+	Status,
+	StatusTags
+} from "common/enums";
+
+export enum TablePillType {
+	VISIBILITY,
+	SPECIES,
+	STATUS
+}
+
+type VisibilityProps = {
+	type: TablePillType.VISIBILITY;
+	value: boolean;
+};
+type SpeciesProps = {
+	type: TablePillType.SPECIES;
+	value: Species;
+};
+type StatusProps = {
+	type: TablePillType.STATUS;
+	value: Status;
+};
+type TablePillProps = VisibilityProps | SpeciesProps | StatusProps;
+
+const TablePill = ({ type, value }: TablePillProps) => {
+	let data: {
+		color: PillColor;
+		text: string;
+	};
+
+	switch (type) {
+		case TablePillType.VISIBILITY:
+			data = {
+				color: value ? PillColor.GREEN : PillColor.RED,
+				text: value ? "yes" : "no"
+			};
+			break;
+		case TablePillType.SPECIES:
+			data = SpeciesTags[value as Species];
```

Comment:
Do we need fallback values like grey for this? I would guess species information would come from the server, and there might be species other than the current 3.

Reply:
for colours, we can enum all available antd colours, but for animals i guess the manual part is that everytime backend is updated with a new animal we would have to update this enum also